.pre-commit-config.yaml

```diff
@@ -52,7 +52,6 @@ repos:
         -   --namespace-packages
         -   --ignore-missing-imports
         -   --warn-redundant-casts
-        -   --warn-unused-ignores
```

Comment:
I had to do this because `pre-commit run` is incorrectly raising "unused ignore" exceptions even though `pre-commit run --all-files` is passing. This prevents commits from passing `pre-commit` checks despite CI passing (because it uses `pre-commit run --all-files`). I can't figure out why the staged files, which should be a proper subset, fail despite the full set passing. Removing this resolves the issue with minimal risk.